eth/beacon/db/chain.py

```diff
@@ -0,0 +1,450 @@
+        and changed "header" to "block".
+        """
+        validate_word(block_hash, title="Block Hash")
+        return block_hash in db
+
+    def persist_block_chain(
+        self,
+        headers: Iterable[BaseBeaconBlock]
+    ) -> Tuple[Tuple[BaseBeaconBlock, ...], Tuple[BaseBeaconBlock, ...]]:
+        """
+        Return two iterable of blocks, the first containing the new canonical blocks,
+        the second containing the old canonical headers
+
+        NOTE: copied from ``eth.db.header.HeaderDB.persist_header_chain``,
+        and changed "header" to "block".
+        """
+        with self.db.atomic_batch() as db:
+            return self._persist_block_chain(db, headers)
+
+    @classmethod
+    def _persist_block_chain(
+            cls,
+            db: BaseDB,
+            blocks: Iterable[BaseBeaconBlock]
+    ) -> Tuple[Tuple[BaseBeaconBlock, ...], Tuple[BaseBeaconBlock, ...]]:
+        """
+
+        NOTE: copied from ``eth.db.header.HeaderDB._persist_header_chain``,
+        and changed "header" to "block".
+        NOTE: moodified a lot
+        """
+
+        try:
+            first_block = first(blocks)
+        except StopIteration:
+            return tuple(), tuple()
+        else:
+            for parent, child in sliding_window(2, blocks):
+                if parent.hash != child.parent_hash:
+                    raise ValidationError(
+                        "Non-contiguous chain. Expected {} to have {} as parent but was {}".format(
+                            encode_hex(child.hash),
+                            encode_hex(parent.hash),
+                            encode_hex(child.parent_hash),
+                        )
+                    )
+
+            is_genesis = first_block.parent_hash == GENESIS_PARENT_HASH
+            if not is_genesis and not cls._block_exists(db, first_block.parent_hash):
+                raise ParentNotFound(
+                    "Cannot persist block ({}) with unknown parent ({})".format(
+                        encode_hex(first_block.hash), encode_hex(first_block.parent_hash)))
+
+            score = 0 if is_genesis else cls._get_score(db, first_block.parent_hash)
+
+        for block in blocks:
+            db.set(
+                block.hash,
+                rlp.encode(block),
+            )
+
+            # FIXME: It's a stub before we implement fork choice rule
+            score += block.slot_number
+
+            db.set(
+                SchemaV1.make_block_hash_to_score_lookup_key(block.hash),
+                rlp.encode(score, sedes=rlp.sedes.big_endian_int),
+            )
+
+        try:
+            previous_canonical_head = cls._get_canonical_head(db).hash
+            head_score = cls._get_score(db, previous_canonical_head)
+        except CanonicalHeadNotFound:
+            (
+                new_canonical_blocks,
+                old_canonical_blocks
+            ) = cls._set_as_canonical_chain_head(db, block.hash)
+        else:
```

Comment:
I think this block (the `try/except` above and the nested `if/else` below would be easier to read written as:

```python
try:
    ...
except CanonicalHeadNotFound:
    return cls._set_as_canonical_chain_head(...)

if score > head_score:
    return cls._set_as_canonical_chain_head(...)
else:
    return tuple(), tuple()
```

Removes a level of nesting and *mostly* decouples the two blocks of code.

Reply:
oops sorry Jason! Wrongly tagged! Asking @pipermerriam

Reply:
Updates:
Sorry, should be `headers[-1]` and it's not a bug because `header` will be set to `headers[-1]` in the previous for loop automatically.